instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java

```diff
@@ -18,6 +18,19 @@
 import org.checkerframework.checker.nullness.qual.Nullable;
 
 // TODO In search for a better home package
+
+/**
+ * Base class for implementing Tracers for HTTP servers. It has 3 types that must be specified by
+ * subclasses:
+ *
+ * @param <REQUEST> - The specific type for HTTP requests
+ * @param <RESPONSE> - The specific type for HTTP responses
+ * @param <CONNECTION> - The specific type of HTTP connection, used to get peer address information
+ *     and HTTP flavor. Use Void if your subclass does not need the connection for peerHostIP,
+ *     peerPort, or flavor.
+ * @param <STORAGE> - Implementation specific storage type for attaching/getting the server context.
```

Comment:
Any doc is better than no doc so this LGTM, but maybe we can go further. `STORAGE` has always been very hard for me to grok. This comment still doesn't really tell me how to use this parameter, "what's server context", "what's implementation specific storage" are still remaining questions.

Reply:
Yeah, I'm planning to work on cleaning up the http server instrumentations after getting through the http clients, and will think on this.